app/auth/strategies/twitter.js

```diff
@@ -0,0 +1,35 @@
+const { Strategy } = require('passport-twitter');
+const { consumerKey, consumerSecret } = require('../../config').oauth.twitter;
+const User = require('../../db/models/user.model');
+const registerUser = require('../../functions/registerUser');
+const generateRandomBytes = require('../../functions/generateRandomBytes');
+
+module.exports = (passport) => {
+  const strategy = new Strategy(
+    {
+      consumerKey,
+      consumerSecret,
+      callbackURL: '/auth/twitter/callback',
+    },
+    async (token, tokenSecret, profile, done) => {
+      try {
+        const user = await User.findOne({ where: { clientID: profile.id } });
+        if (user) {
+          return done(null, user);
+        }
+        registerUser(
+          profile.id, // FIXME: Find a better solution than this
```

Comment:
Let's find a better solution now - What are our options?

Reply:
Since the Twitter API doesn't return an email, we are kinda screwed. We could McGyver a ridiculous string, but I don't think that would be plausible. I'd rather replace the Twitter strategy with Facebook or ditch 'em both if we are still not going to use it.